integration.js

```
@@ -139,23 +180,11 @@ function _lookupEntity(entityObj, options, cb) {
 
   log.trace({ cql }, 'CQL Request Parameter');
 
-  requestWithDefaults(requestOptions, function (err, response, body) {
-    // check for a request error
-    if (err) {
-      cb({
-        detail: 'Error Making HTTP Request',
-        debug: err
-      });
-      return;
-    }
+  requestWithDefaults(requestOptions, function (error, response, body) {
+    const restErr = handleRestError(response, options, requestOptions, error);
 
-    // If we get a 404 then cache a miss
-    if (response.statusCode === 404) {
-      cb(null, {
-        entity: entityObj,
-        data: null // setting data to null indicates to the server that this entity lookup was a "miss"
-      });
-      return;
+    if (restErr) {
+      return cb(restErr);
     }
 
     if (response.statusCode === 400) {
```

Should 400 errors be handled as a miss here?  What does confluence return if no results are found.  I'm wondering if we should also handle this 400 error up in the new error handling logic and treat it as an error.  

It's possible that we were getting random 400 errors though and decided to just "swallow" those errors.  Have you seen this happen at all during your development?

---

Looking at the documentation,  I don't think the 400s should be handled as a miss - the API will return a 200 for missed results, with an empty array as the body value. I did not run into any random 400 errors while working on this.

The change I'm committing will return an error for 400s.